framework/tlvf/src/src/WSC/AttrList.cpp

```diff
@@ -208,10 +208,11 @@ bool AttrList::init()
                 return false;
             }
             break;
-        // Other attributes are not expected, if so ignore them
+        // Other attributes are not expected, if so ignore them silently
         default:
-            TLVF_LOG(ERROR) << "Unknown attribute " << getNextAttrType();
-            break;
+            TLVF_LOG(DEBUG) << "Unknown attribute " << getNextAttrType()
+                            << " assuming end of AttrList";
+            return true;
```

Comment:
There really is no reason why an unkown attribute should be at the end of the attrlist...

However, the WSC specification is pretty strict. There simply shouldn't be an unknown attribute. If there is, it's an error. So I think a practical approach would be to return false here. Or did you see a certification device with an unknown attribute?

Reply:
No, there should not be any unknown attributes.

The problem is that we pass a buffer size which is larger than the size of the config data. So the while loop will eventually reach the end of the buffer, which has "garbage" data. So I actually used it to find the end of the list.